### DISCUSSION TITLE
Moving a complex diagram around

### DISCUSSION THREAD

**Discussion by blackdust21** [excalidraw/excalidraw#5341] [General]
I'm new to excalidraw and I find myself using it more and more. I have searched around and was not able to find much help in my question.
Now I'm drawing complex block diagrams. But how do I move my diagrams around?
When I click on a block, I can only move it.
Using Shift+Click is not very good as one has to click every single block, pls the text. And it's very buggy. One wrong click and it will un-click everything thus wasting your efforts in selecting the components.
SO is there a drag to select and then move around? If not, can it be implemented?

Move around: Meaning moving diagram from one location on canvas to another.

**Comment by sojusnik** [comment]
Left mouse click and hold it so select the drawings you want and then move them as a group whenever you want.

**Reply by dwelle** [reply]
Yep. @blackdust21 But I assume you've tried this and it didn't work? What device are you using?

![excalidraw-box-select](https://user-images.githubusercontent.com/5153846/175091803-abd6c8c1-7873-4a45-a438-f1ec010d60cf.gif)
